batch/batches/Archive/KAsyncLiveEntryArchive.class.php

```diff
@@ -7,79 +7,89 @@
-    protected function exec(KalturaBatchJob $job)
-    {
-        $jobData = $job->data;
-        /** @var KalturaLiveEntryArchiveJobData $jobData*/
-        $liveEntryId = $jobData->liveEntryId;
-        $liveEntry = KBatchBase::$kClient->baseEntry->get($liveEntryId);
-        /** @var KalturaLiveStreamEntry $liveEntry */
-        $notDeletedCuePointTags = $liveEntry->recordingOptions->nonDeletedCuePointsTags;
-        $this->deleteCuePoints($liveEntryId, $notDeletedCuePointTags);
-
-        $vodEntryId = $jobData->vodEntryId;
-        $vodEntry = KBatchBase::$kClient->baseEntry->get($vodEntryId);
-        $this->updateEntriesData($liveEntry, $vodEntry);
-
-        $this->clearPushNotificationQueue($liveEntryId, $liveEntry->partnerId);
-
-        return $this->closeJob($job, null, null, "Auto archive finished", KalturaBatchJobStatus::FINISHED);
-    }
-
-    protected function clearPushNotificationQueue($liveEntryId, $partnerId)
-    {
-        $liveEntryString = new KalturaStringValue();
-        $liveEntryString->value = $liveEntryId;
-        $pushNotificationParameter = new KalturaPushEventNotificationParameter();
-        $pushNotificationParameter->key = "entryId";
-        $pushNotificationParameter->value = $liveEntryString;
-        $pushEventNotificationParameterArray = array();
-        $pushEventNotificationParameterArray[] = $pushNotificationParameter;
-        $pushNotificationParams = new KalturaPushNotificationParams();
-        $pushNotificationParams->userParams = $pushEventNotificationParameterArray;
-
-        KBatchBase::impersonate($partnerId);
-        KBatchBase::$kClient->eventNotificationTemplate->sendCommand(self::POLLS_PUSH_NOTIFICATIONS_STRING,
-            $pushNotificationParams, KalturaPushNotificationCommandType::CLEAR_QUEUE);
-        KBatchBase::$kClient->eventNotificationTemplate->sendCommand(self::PUBLIC_QNA_NOTIFICATIONS_STRING,
-            $pushNotificationParams, KalturaPushNotificationCommandType::CLEAR_QUEUE);
-        KBatchBase::$kClient->eventNotificationTemplate->sendCommand(self::USER_QNA_NOTIFICATIONS_STRING,
-            $pushNotificationParams, KalturaPushNotificationCommandType::CLEAR_QUEUE);
-        KBatchBase::unimpersonate();
-    }
+	/* (non-PHPdoc)
+	 * @see KBatchBase::getType()
+	 */
+	public static function getType()
+	{
+		return KalturaBatchJobType::LIVE_ENTRY_ARCHIVE;
+	}
+
+	/**
+	 * (non-PHPdoc)
+	 * @see KBatchBase::getJobType()
+	 */
+	protected function getJobType()
+	{
+		return KalturaBatchJobType::LIVE_ENTRY_ARCHIVE;
+	}
+
+	/**
+	 * @param KalturaBatchJob $job
+	 * @return KalturaBatchJob
+	 */
+	protected function exec(KalturaBatchJob $job)
+	{
+		$jobData = $job->data;
+		/** @var KalturaLiveEntryArchiveJobData $jobData*/
+		$liveEntryId = $jobData->liveEntryId;
+		$liveEntry = KBatchBase::$kClient->baseEntry->get($liveEntryId);
+		/** @var KalturaLiveStreamEntry $liveEntry */
+		if (!$liveEntry){
+			KalturaLog::err("Live entry [" . $liveEntryId . "] not found");

```

Comment:
if this is the case I think we should close the job with a different status than finished and a different error message
"Could not get live entry with id <live_entry_id>"
Status should be KalturaBatchJobStatus::FAILED